.github/workflows/build.yaml

```diff
@@ -89,39 +72,60 @@ jobs:
       uses: actions/checkout@v3
       with:
         fetch-depth: 1
-    - name: Installing dependencies
-      run: sh ./.github/scripts/install_deps.sh
-    - name: CMake
-      run: |
-        # Prepare build dir
-        echo "Will use $(nproc) cores."
-        mkdir build
-        pushd build
-        cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ..
-        popd
-    - name: Restore cache
-      uses: actions/cache/restore@v3
-      with:
-        path: build/clang-tidy-cache
-        key: clang-tidy-cache-${{ github.sha }}
-        restore-keys: |
-          clang-tidy-cache-
-    - name: Check
-      # To run a single check, call:
-      # python3 ../utils/run-clang-tidy.py -checks=-*,my-check-prefix* | tee ../clang-tidy.log
-      run: |
-        # run-clang-tidy.py
-        pushd build
-        python3 ../utils/run-clang-tidy.py -cache | tee ../clang-tidy.log
-        popd
-    - name: Save cache
-      # We want to save the cache even if the report fails because of failed files
-      uses: actions/cache/save@v3
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
       with:
-        path: build/clang-tidy-cache
-        key: clang-tidy-cache-${{ github.sha }}
-    - name: Report
-      run: utils/check_clang_tidy_results.py clang-tidy.log
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          clang_tidy:
+            - 'src/**'
+            - 'utils/check_clang_tidy_results.py'
+            - 'utils/run-clang-tidy.py'
+          compile: &compile
+            - 'CMakeLists.txt'
+            - 'cmake/**'
+            - 'src/**'
+          test_suite:
+            - *compile
+            - 'data/**'
+            - 'regression_test.py'
+            - 'test/**'
+            - '.github/workflows/build_testsuite.yaml'
+          installer:
+            - *compile
+            - 'data/**'
+            - 'doc/**'
+            - 'po/**'
+          install_dependencies: &install_dependencies
+            - 'install-dependencies.sh'
+            - 'utils/download_asio.sh' # called by install-dependencies.sh
+          inst_windows_msvc:
+            - *install_dependencies
+            - 'utils/win32/**'
+            - '.github/scripts/vcpkg_ref'
+            - '.github/workflows/build_windows_msvc.yaml'
+          inst_windows:
+            - *install_dependencies
+            - 'utils/win32/**'
+            - '.github/workflows/build_windows.yaml'
+          inst_macos:
+            - 'compile.sh' # called by utils/macos/build_app.sh
+            - *install_dependencies
+            - 'utils/macos/*'
+            - '.github/workflows/build_macos.yaml'
+          appimage:
+            - 'xdg/org.widelands.Widelands.*'
+            - '.github/workflows/build_appimage.yaml'
+    - name: Clang-tidy steps
+      if: steps.changed_files.outputs.clang_tidy == 'true'
+      uses: ./.github/actions/build_clang_tidy.yml
```

Comment:
I actually don't quite understand why you treat clang-tidy differently and why you put it in a separate directory.

Reply:
This can be changed if the checking is a separate job, what you suggest. Because it just are some steps, it is a "composite action" and not a "workflow" now.

Reply:
Is changed.